FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/TestUtils.cs

```diff
@@ -99,5 +107,23 @@ public static IDataSchemaConverter TestDataSchemaConverterDelegate(string name)
                     NullLogger<CustomSchemaConverter>.Instance);
             }
         }
+
+        public static IFhirSpecificationProvider TestFhirSpecificationProviderDelegate(FhirVersion fhirVersion)
+        {
+            var dataClient = Substitute.For<IFhirDataClient>();
+
+            var metadataOptions = new MetadataOptions();
+            dataClient.Search(metadataOptions)
+                .ReturnsForAnyArgs(x => TestDataProvider.GetBundleFromFile(TestDataConstants.R4MetadataFile));
+
+            if (fhirVersion == FhirVersion.R4)
```

Comment:
The `if` and `else` act identically, is it by design?

Reply:
This is just for test, in actual delegate we use **switch** and will throw exceptions if fhirVersion is undefined.